File: 프로그래머스/lv1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.java
import java.util.*;
class Solution {
    public int[] solution(int[] numbers) {
        int[] answer = {};
        Set<Integer> plusSet = new HashSet<>();
        for(int i = 0; i < numbers.length - 1; i++){
            for(int j = i+1; j <numbers.length; j++){
                plusSet.add(numbers[i] + numbers[j]);
            }
        }
        answer = new int[plusSet.size()];
        Iterator<Integer> iter = plusSet.iterator();
        int idx = 0;
        while(iter.hasNext()){
            answer[idx++] = iter.next();
        }
        Arrays.sort(answer);
        return answer;
    }
}

File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.js
function solution(numbers) {
    let allPossibleSums = [];
    for (let i = 0; i < numbers.length; i++){
        for (let j = i + 1; j < numbers.length; j++){
            allPossibleSums.push(numbers[i] + numbers[j]);
        }
    }
    
    return [...new Set(allPossibleSums)].sort((a, b) => a - b);
}